script/generate_needle_previews

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/perl -w
+# Copyright 2025 Roland Clobus <rclobus@rclobus.nl>
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+
+=head1 generate_needle_previews
+
+generate_needle_previews - generate .svg files with the needle information as an overlay
+
+=head1 SYNOPSIS
+
+Use a list of files (using stdin or arguments):
+
+  ls -1 *.json | generate_needle_previews -
+  generate_needle_previews *.json
+
+Look for a specific tag:
+
+  grep "DESKTOP-gnome" -l *.json | generate_needle_previews -
+
+=cut
+
+use Mojo::Base -strict, -signatures;
+use Getopt::Long;
+use Mojo::JSON qw(decode_json);
+use Mojo::File qw(path);
+use Mojo::Util qw(url_escape);
+
+sub process_file ($filename) {
+    my $SVGFILE;
+
+    chomp $filename;
+    my $pp = Mojo::File->new($filename);
+    my $name = $pp->basename(".json");
+    printf "Working on %s\n", $name;
+    my $pngpath = path($pp->dirname, "${name}.png");
+    my $output = qx/file --brief ${pngpath}/;
+    my ($width, $height) = $output =~ /^PNG image data, (\d+) x (\d+)/;
+    if (!$width or !$height) {
+        say 'E: Could not determine image size';
+        return
+    }
+
+    my $data  = decode_json($pp->slurp);
+    my @areas = @{$data->{area}};
```

Comment:
There's no need to make a copy:
```suggestion
    my $areas = $data->{area};
```
With this you of course need to change the code below from `@areas` to `@$areas`. Maybe you can also just replace `@areas` below with `@{$data->{area}}` and remove this line completely.